range_queries/mo.cpp

```diff
@@ -1,50 +1,84 @@
+/**
+ *
+ * @file
+ * \brief A C++ Program to compute unique elements of ranges for different range
+ * queries
+ *
+ * \details
+ * The preprocessing part takes O(m Log m) time.
+ * Processing all queries takes O(n * √n) + O(m * √n) = O((m+n) * √n) time.
+ */
+#include <algorithm>
 #include <iostream>
-using namespace std;
 const int N = 1e6 + 5;
 int a[N], bucket[N], cnt[N];
+// Variable to represent block size. This is made global so compare() of sort
+// can use it.
```

Comment:
Use a comment structure **except** inside functions.
Same in various other parts of the code.

Reply:
```
/*

*/
```

you mean to say this way.

Reply:
Yes, that would be correct. 👍